lib/link_checker.rb

```diff
@@ -0,0 +1,46 @@
+require 'yaml'
+require 'rest-client'
+require_relative 'root'
+require_relative 'app_config'
+require_relative 'url_checker'
+
+
+
+class LinkChecker
+  def initialize
+    @urls = YAML.load_file(GOVUK_PAGES_FILE)
```

Comment:
Sorry Jack, I disagree.  In OO programming, the whole point is to hide internal variables from the outside world unless they absolutely must have access to them.  So we declare all our instance variables prefixed with a @, and then just list the ones that are read or written from the outside world.  It's immediately obvious what is private data, and what is exposed to the outside world.  

If you write attr_reader and attr_accessor for them all, this distinction disappears.  Writing attr_reader and attr_accessor then making them private seems to be like scratching your right ear with your left foot - you're going a long way round to do something which the language gives you for free.